src/result.ts

```diff
@@ -71,6 +72,7 @@ export function valueResultFromProto(result: proto.StmtResult, intMode: IntMode)
     const stmtResult = stmtResultFromProto(result);
     let value: Value | undefined;
     if (result.rows.length > 0 && stmtResult.columnNames.length > 0) {
+        // TODO: How do we solve this? AFAICS we don't have column data when fetching a single value, so we don't know when to cast ints to booleans
```

Comment:
This is the biggest drawback of the whole PR I think. Since we don't get any column type data when fetching pure values, we can't know when we should cast. We'd have to make it very clear that the casting config only applies to query/queryRow and not when fetching values. Thoughts?